File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    
    arr1 = [bin(i)[2:] for i in arr1]
    arr2 = [bin(i)[2:] for i in arr2]
    
    for i in range(len(arr1)):
        if len(arr1[i]) != n:
            while (n - len(arr1[i])):
                arr1[i] = '0' + arr1[i]
            arr1[i] = arr1[i]
    
    for i in range(len(arr2)):
        if len(arr2[i]) != n:
            while (n - len(arr2[i])):
                arr2[i] = '0' + arr2[i]
            arr2[i] = arr2[i]
    
    map = [[0]*len(arr1) for i in range(len(arr1))]
    
    for i in range(len(arr1)):
        for j in range(len(arr1)):
            if (arr1[i][j] == '1') or (arr2[i][j] == '1'):
                map[i][j] = '#'
            else:
                map[i][j] = ' '
    
    print(map)
    return [''.join(i) for i in map]

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
   public static String[] solution(int n, int[] arr1, int[] arr2) {
		String[] answer = new String[n];

		for (int i = 0; i < arr1.length; i++) {
			String binary1 = String.format("%" + n + "s", Integer.toBinaryString(arr1[i])).replace(' ', '0');
			String binary2 = String.format("%" + n + "s", Integer.toBinaryString(arr2[i])).replace(' ', '0');

			StringBuilder tmp = new StringBuilder();

			for (int j = 0; j < binary1.length(); j++) {
				if (binary1.charAt(j) == '1' || binary2.charAt(j) == '1') {
					tmp.append('#');
				} else {
					tmp.append(' ');
				}
			}

			answer[i] = tmp.toString();
		}
		return answer;
	}
}